app/controllers/text_books_controller.rb

```diff
@@ -5,8 +5,9 @@ def index
       @q = TextBook.where(genre: ["Basic", "Git", "Ruby", "Ruby on Rails"]).ransack(params[:q])
       @text_books = @q.result.order(:id)
     else
-      @text_books = TextBook.where(genre: params[:genre])
+      @q = TextBook.where(genre: params[:genre]).ransack(params[:q])
     end
+    @text_books = @q.result
```

Comment:
6行目についているorderをこちらに反映させて下さい。

```suggestion
    @text_books = @q.result.order(:id)
```